airflow/providers/amazon/aws/hooks/elasticache.py

```diff
@@ -0,0 +1,259 @@
+    RESPONSE_MODIFYING_STATE = 'has status modifying which is not valid for deletion'
+
+    # Constants for ElastiCache replication group status
+    STATUS_CREATING = 'creating'
+    STATUS_AVAILABLE = 'available'
+    STATUS_DELETING = 'deleting'
+    STATUS_CREATE_FAILED = 'create-failed'
+
+    def __init__(
+        self, max_retries=10, exponential_back_off_factor=1, initial_poke_interval=60, *args, **kwargs
+    ):
+        """
+        :param max_retries: Max tries for checking availability of and deleting replication group
+        :param exponential_back_off_factor: Factor for deciding next sleep time
+        :param initial_poke_interval: Initial sleep time in seconds
+        """
+        self.max_retries = max_retries
+        self.exponential_back_off_factor = exponential_back_off_factor
+        self.initial_poke_interval = initial_poke_interval
+
+        super().__init__(client_type='elasticache', *args, **kwargs)
+
+    def create_replication_group(self, config):
+        """
+        Call ElastiCache API for creating a replication group
+        :param config: Python dictionary to use as config for creating replication group
+        :return: Response from ElastiCache create replication group API
+        """
+        return self.conn.create_replication_group(**config)
+
+    def delete_replication_group(self, replication_group_id):
+        """
+        Call ElastiCache API for deleting a replication group
+        :param replication_group_id: ID of replication group to delete
+        :return: Response from ElastiCache delete replication group API
+        """
+        return self.conn.delete_replication_group(ReplicationGroupId=replication_group_id)
+
+    def describe_replication_group(self, replication_group_id):
+        """
+        Call ElastiCache API for describing a replication group
+        :param replication_group_id: ID of replication group to describe
+        :return: Response from ElastiCache describe replication group API
+        """
+        return self.conn.describe_replication_groups(ReplicationGroupId=replication_group_id)
+
+    def get_replication_group_status(self, replication_group_id):
+        """
+        Get current status of replication group
+        :param replication_group_id: ID of replication group to check for status
+        :return: Current status of replication group
+        """
+        return self.describe_replication_group(replication_group_id)[self.REPLICATION_GROUPS][0][self.STATUS]
+
+    def is_replication_group_available(self, replication_group_id):
+        """
+        Helper for checking if replication is available or not
+        :param replication_group_id: ID of replication group to check for availability
+        :return: True if available else False
+        replication group
+        """
+        return self.get_replication_group_status(
+            replication_group_id=replication_group_id
+        ) == self.STATUS_AVAILABLE
```

Comment:
```suggestion
        return self.get_replication_group_status(replication_group_id) == self.STATUS_AVAILABLE
```

It's rather needlessly verbose to use a named argument for a function with only takes one parameter.